components/org.wso2.carbon.identity.scim2.provider/src/main/java/org/wso2/carbon/identity/scim2/provider/resources/GroupResource.java

```diff
@@ -312,59 +312,76 @@ private Response processRequest(final Map<String, String> requestAttributes) {
         String attributes = requestAttributes.get(SCIMProviderConstants.ATTRIBUTES);
         String excludedAttributes = requestAttributes.get(SCIMProviderConstants.EXCLUDE_ATTRIBUTES);
         String search = requestAttributes.get(SCIMProviderConstants.SEARCH);
-
         JSONEncoder encoder = null;
         try {
-
             IdentitySCIMManager identitySCIMManager = IdentitySCIMManager.getInstance();
-            //obtain the encoder at this layer in case exceptions needs to be encoded.
+            // Obtain the encoder at this layer in case exceptions needs to be encoded.
             encoder = identitySCIMManager.getEncoder();
 
-            // obtain the user store manager
+            // Obtain the user store manager
             UserManager userManager = IdentitySCIMManager.getInstance().getUserManager();
 
-            //create charon-SCIM group endpoint and hand-over the request.
+            // Create charon-SCIM group endpoint and hand-over the request.
             GroupResourceManager groupResourceManager = new GroupResourceManager();
             SCIMResponse scimResponse = null;
-            int startIndex = 0;
-            int count = 0;
             if (GET.class.getSimpleName().equals(httpVerb) && id == null) {
                 String filter = requestAttributes.get(SCIMProviderConstants.FILTER);
-                if(requestAttributes.get(SCIMProviderConstants.START_INDEX) != null){
-                    startIndex = Integer.parseInt(requestAttributes.get(SCIMProviderConstants.START_INDEX));
-                }
-                if(requestAttributes.get(SCIMProviderConstants.COUNT) != null){
-                    count = Integer.parseInt(requestAttributes.get(SCIMProviderConstants.COUNT));
-                }
-
                 String sortBy = requestAttributes.get(SCIMProviderConstants.SORT_BY);
                 String sortOrder = requestAttributes.get(SCIMProviderConstants.SORT_ORDER);
                 String domainName = requestAttributes.get(SCIMProviderConstants.DOMAIN);
 
-                scimResponse = groupResourceManager.listWithGET(userManager, filter, startIndex,
-                        count, sortBy, sortOrder, domainName, attributes, excludedAttributes);
-
+                // Processing count and startIndex in the request.
+                Integer startIndex = processPaginationParamsInRequest(
```

Comment:
Does the `processPaginationParamsInRequest` method is itself process only the integer parameters? if so shall we change the method name to reflect that?. And since this method process only a single parameter, better to reflect that as well

Reply:
Fixed: df77256698c74c829147f73d00e2f63979199c16